src/HeaderCell.tsx

```diff
@@ -93,7 +82,7 @@ export default function HeaderCell<R, K extends keyof R>({
         column={column}
         onResize={props.onResize}
       >
-        {cell}
+        {cell as React.ReactElement<React.ComponentProps<'div'>>}
```

Comment:
May be returning `<>column.name</>` is better if TS is not recognizing the string

Reply:
I think the issue is that we're assigning various types to `cell`, and TS doesn't understand that it's a div element at this moment, so I think the type assertion is fair here.